notebook/static/notebook/js/notebook.js

```diff
@@ -164,29 +164,79 @@ import {ShortcutEditor} from 'notebook/js/shortcuteditor';
         var that = this;
 
         Object.defineProperty(this, 'line_numbers', {
-          get: function(){
-            var d = that.config.data||{}
-            var cmc =  (d['Cell']||{})['cm_config']||{}
-            return cmc['lineNumbers'] || false;
-          },
-          set: function(value){
-            this.get_cells().map(function(c) {
-              c.code_mirror.setOption('lineNumbers', value);
-            })
-            that.config.update({'Cell':{'cm_config':{'lineNumbers':value}}})
-          }
-          
-        })
+            get: function() {
+                var d = that.config.data || {};
+                var cmc =  (d['Cell'] || {}) ['cm_config'] || {};
+                return cmc['lineNumbers'] || false;
+            },
+            set: function(value) {
+                this.get_cells().map(function(c) {
+                    c.code_mirror.setOption('lineNumbers', value);
+                });
+                that.config.update({
+                    'Cell': {
+                        'cm_config': {
+                            'lineNumbers':value
+                        }
+                    }
+                });
+            }
+        });
+        
+        Object.defineProperty(this, 'header', {
+            get: function() {
+                return that.class_config.get_sync('Header');
+            },
+            set: function(value) {
+                if (value === true) {
+                    $('#header-container').show();
+                    $('.header-bar').show();
+                } else if (value === false) {
+                    $('#header-container').hide();
+                    $('.header-bar').hide();
+                }
+                that.events.trigger('resize-header.Page');
+                that.class_config.set('Header', value);
+            }
+        });
+                
+        Object.defineProperty(this, 'toolbar', {
+            get: function() {
+                return that.class_config.get_sync('Toolbar');
+            },
+            set: function(value) {
+                if (value === true) {
+                    $('div#maintoolbar').show();
+                } else if (value === false) {
+                    $('div#maintoolbar').hide();
+                }
+                that.events.trigger('resize-header.Page');
+                that.class_config.set('Toolbar', value);
+            }
+        });
+        
+        this.class_config.get('Header').then(function(header) {
+            if (header === false) {
+                that.header = false;
```

Comment:
This is a good point, thanks for the feedback. I just pushed a new commit that moves the toggle logic to the actions and then calls the actions to toggle the header/toolbar on notebook load.